core/src/main/java/org/springframework/ldap/core/SpringLdapClient.java

```diff
@@ -0,0 +1,666 @@
+	 */
+	public void setIgnoreSizeLimitExceededException(boolean ignoreSizeLimitExceededException) {
+		this.ignoreSizeLimitExceededException = ignoreSizeLimitExceededException;
+	}
+
+	private final class SpringListSpec implements LdapClient.ListSpec {
+		private final Name name;
+
+		private SpringListSpec(Name name) {
+			this.name = name;
+		}
+
+		@Override
+		public <T> List<T> toList(NameClassPairMapper<T> mapper) {
+			ContextExecutor<NamingEnumeration<NameClassPair>> executor = (ctx) -> ctx.list(this.name);
+			NamingEnumeration<NameClassPair> results = computeWithReadOnlyContext(executor);
+			return SpringLdapClient.this.toList(results, mapper::mapFromNameClassPair);
+		}
+
+		@Override
+		public <T> Stream<T> toStream(NameClassPairMapper<T> mapper) {
+			ContextExecutor<NamingEnumeration<NameClassPair>> executor = (ctx) -> ctx.list(this.name);
+			NamingEnumeration<NameClassPair> results = computeWithReadOnlyContext(executor);
+			return SpringLdapClient.this.toStream(results, mapper::mapFromNameClassPair);
+		}
+	}
+
+	private final class SpringListBindingsSpec implements LdapClient.ListBindingsSpec {
+		private final Name name;
+
+		private SpringListBindingsSpec(Name name) {
+			this.name = name;
+		}
+
+		@Override
+		public <T> List<T> toList(NameClassPairMapper<T> mapper) {
+			ContextExecutor<NamingEnumeration<Binding>> executor = (ctx) -> ctx.listBindings(this.name);
+			NamingEnumeration<Binding> results = computeWithReadOnlyContext(executor);
+			return SpringLdapClient.this.toList(results, mapper::mapFromNameClassPair);
+		}
+
+		@Override
+		public <T> List<T> toList(ContextMapper<T> mapper) {
+			ContextExecutor<NamingEnumeration<Binding>> executor = (ctx) -> ctx.listBindings(this.name);
+			NamingEnumeration<Binding> results = computeWithReadOnlyContext(executor);
+			return SpringLdapClient.this.toList(results, function(mapper));
+		}
+
+		@Override
+		public <T> Stream<T> toStream(NameClassPairMapper<T> mapper) {
+			ContextExecutor<NamingEnumeration<Binding>> executor = (ctx) -> ctx.listBindings(this.name);
+			NamingEnumeration<Binding> results = computeWithReadOnlyContext(executor);
+			return SpringLdapClient.this.toStream(results, mapper::mapFromNameClassPair);
+		}
+
+		@Override
+		public <T> Stream<T> toStream(ContextMapper<T> mapper) {
+			ContextExecutor<NamingEnumeration<Binding>> executor = (ctx) -> ctx.listBindings(this.name);
+			NamingEnumeration<Binding> results = computeWithReadOnlyContext(executor);
+			return SpringLdapClient.this.toStream(results, function(mapper));
+		}
+	}
+
+	private final class SpringAuthenticateSpec implements LdapClient.AuthenticateSpec {
+		LdapClient.SearchSpec search = new SpringSearchSpec();
+
+		String password;
+
+		@Override
+		public LdapClient.AuthenticateSpec query(LdapQuery query) {
+			this.search.query(query);
+			return this;
+		}
+
+		@Override
+		public LdapClient.AuthenticateSpec password(String password) {
+			this.password = password;
```

Comment:
Since I don't know much about existing support for this feature, I thought I'd ask. Does anything special need to happen to protect or erase this password after use?